feature/s3/manager/download.go

```diff
@@ -401,6 +405,9 @@ func (d *downloader) tryDownloadChunk(params *s3.GetObjectInput, w io.Writer) (i
 		return 0, err
 	}
 	d.setTotalBytes(resp) // Set total if not yet set.
+	d.once.Do(func() {
+		d.etag = aws.ToString(resp.ETag)
```

Comment:
is eTag guaranteed to appear on the response? otherwise we will panic when doing `aws.ToString`

Reply:
Manually get object with/without version, and etag will always be returned even if the object is empty. `aws.ToString` would nil check and return empty string for nil input